app/webpacker/components/ContextMenu.vue

```diff
@@ -69,27 +69,36 @@
 			addScrollEventListener () {
 				window.addEventListener('scroll', this.close);
 			},
-            onFocusout (e) {
-              if(e.relatedTarget != this.$el &&
-                 !this.$el.contains(e.relatedTarget)){
-                this.close(true);
-              }
-            },
-            /**
-             * Close the context menu.
-             *
-             * @param {boolean|Event} emit Used to prevent event being emitted twice from when menu is clicked and closed
-             */
-            close (emit = true) {
-                // return;
-                this.top = null;
-                this.left = null;
-                this.data = null;
-                this.show = false;
-                if (emit) {
-                    this.$emit('close');
-                }
-            },
+      onFocusout (e) {
+        if(e.relatedTarget != this.$el &&
+           !this.$el.contains(e.relatedTarget)){
+            this.close(true);
+
+            // When clicking away from note or link annotation during creation, destroy temporarily stored annotation in store 
+            let annotation = this.$store.getters['annotations/getById'](this.$tempId);
```

Comment:
@cgruppioni I would leave knowledge of annotations out of ContextMenu so it's usable as a general UI element and instead handle this sort of thing in either a `close` handler that's passed in or modify the ContextMenu component to accept an `onFocusout` handler if it needs to be handled on that event specifically.